### PR TITLE
feat(app): Add shadow camera helpers to v-light-helper

### PR DESCRIPTION
**Pull request by JaimeTorrealba**

### CODE REVIEW

File: src/core/directives/vLightHelper.ts [hunk 1]
```diff
@@ -1,36 +1,66 @@
...
   },
 }
 let currentHelper: any
 let currentInstance: any
+let currentShadowCameraHelper: any
+
+const getHelperIntance = (el: any) =>
+  el.parent.children.find((child: any) => child instanceof currentHelper)
+const getShadowCameraHelperIntance = (el: any) =>
+  el.parent.children.find((child: any) => child instanceof CameraHelper)
```
Comment (andretchen0) [inline src/core/directives/vLightHelper.ts:62]: I think this approach is causing some problems. If you `el.parent.children.find` there might be multiple instances of a particular `instanceof`, if there are sibling lights.

## Suggestion

LightHelpers and CameraHelpers store a reference to their targets. You could match on those.

File: src/core/directives/vLightHelper.ts [hunk 2]
```diff
@@ -1,36 +1,66 @@
...
+    // if has ShadowCamera
+    if (!currentShadowCameraHelper) return
+    currentShadowCameraHelper = getShadowCameraHelperIntance(el)
+    currentShadowCameraHelper.dispose()
+    el.parent.remove(currentShadowCameraHelper)
   },
 }
 let currentHelper: any
 let currentInstance: any
+let currentShadowCameraHelper: any
```
Comment (andretchen0) [inline src/core/directives/vLightHelper.ts:57]: I think these variables are out of sync when the code reaches line 29. Iirc:

* Create SpotLight 1, add helper
* Create PointLight 2, add helper
* updated() on Light 1 <-- `getHelperInstance` will return the PointLight helper instance

## Suggestion

If you want to store references, maybe something like this?

```
type LightHelper = DirectionalLightHelper | PointLightHelper | SpotLightHelper | HemisphereLightHelper | RectAreaLightHelper
const lightUuidToShadowCameraHelper: Record<string, CameraHelper> = {}
const lightUuidToLightHelper: Record<string, LightHelper> = {}
```

**Review by andretchen0 — CHANGES_REQUESTED** [review]
Hey @JaimeTorrealba !

I've found some bugs and made a few suggestions for getting rid of them.

There are also a few changes/suggestions for typos, etc.

😎

**Comment by JaimeTorrealba** [comment]
@andretchen0 thanks for taking the time, I like your feedback. I'm going to put this in draw because we're moving the directive section to the core pkg. So I don't know yet what will happen with this one

**Comment by JaimeTorrealba** [comment]
I'll close this one, since now the directives are in the core (I'll re-open there)

**Pull request closed without merging**